master/custom/release_dashboard.py

```diff
@@ -1,122 +1,730 @@
 import datetime
 import os
 import time
+from functools import cached_property, total_ordering
+import enum
+from dataclasses import dataclass
+import itertools
+import urllib.request
+import urllib.error
+import json
+from pathlib import Path
+from xml.etree import ElementTree
 
 from flask import Flask
 from flask import render_template, request
+import jinja2
+import humanize
 
 from buildbot.data.resultspec import Filter
+import buildbot.process.results
+
+N_BUILDS = 200
+MAX_CHANGES = 50
 
-FAILED_BUILD_STATUS = 2
 
 # Cache result for 6 minutes. Generating the page is slow and a Python build
 # takes at least 5 minutes, a common build takes 10 to 30 minutes.  There is a
 # cronjob that forces a refresh every 5 minutes, so all human requests should
 # get a cache hit.
 CACHE_DURATION = 6 * 60
 
+BRANCHES_URL = "https://raw.githubusercontent.com/python/devguide/main/include/release-cycle.json"
 
-def get_release_status_app(buildernames):
-    release_status_app = Flask("test", root_path=os.path.dirname(__file__))
-    buildernames = set(buildernames)
-    cache = None
 
-    def get_release_status():
-        builders = release_status_app.buildbot_api.dataGet("/builders")
+def _gimme_error(func):
+    """Decorator to turn AttributeError into a different Exception
 
-        failed_builds_by_branch_and_tier = {}
-        disconnected_workers = {}
+    jinja2 tends to swallow AttributeError or report it in some place it
+    didn't happen. When that's a problem, use this decorator to get
+    a usable traceback.
+    """
+    def decorated(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except AttributeError as e:
+            raise Exception(f'your error: {e!r}')
```

Comment:
Maybe raise something more concrete (maybe `TemplateAttributeError` or something else).

Reply:
Sure.
(This is a debug helper -- but one that I find myself using constantly so I'd rather leave it in.)